### PR TITLE
elm: add patch to build on cabal-install-3.2

### PR DESCRIPTION
**Pull request by fishtreesugar**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
#67284

### CODE REVIEW

File: Formula/elm.rb [hunk 1]
```diff
@@ -18,10 +18,14 @@ class Elm < Formula
   uses_from_macos "ncurses"
   uses_from_macos "zlib"
 
-  def install
-    # elm-compiler needs to be staged in a subdirectory for the build process to succeed
-    (buildpath/"elm-compiler").install Dir["*"]
+  patch do
+    # elm's tarball is not a proper cabal tarball, it contains multiple cabal files.
+    # Add `cabal.project` lets cabal-install treat this tarball as cabal project correctly.
+    url "https://github.com/fishtreesugar/compiler/commit/eb566e901a419a6620e43c18faf89f57f0827124.patch?full_index=1"
```
Comment (SeekingMeaning) [inline Formula/elm.rb:24]:
```suggestion
    # https://github.com/elm/compiler/pull/<insert PR number here>
    url "https://github.com/elm/compiler/commit/eb566e901a419a6620e43c18faf89f57f0827124.patch?full_index=1"
```

Comment (SeekingMeaning) [reply Formula/elm.rb:24]: Nice work! Could you submit this as a PR to upstream?

Comment (fishtreesugar) [reply Formula/elm.rb]: done

**Review by SeekingMeaning — DISMISSED** [review]
Thanks, @fishtreesugar!

**Review by SeekingMeaning — APPROVED** [review]

**Pull request closed without merging**